File: remove_duplicates_from_sorted_array_2.py
#0(N^2) TIME AND O(1) SPACE
# THIS IS BRUTE FORCE APPROACH
class Solution:
    def removeDuplicates(self, nums):
        if len(nums) == 0:
            return 0
        last_seen = nums[0]
        last_seen_count = 1
        i = 1
        while i < len(nums):
            num = nums[i]
            if num == last_seen:
                if last_seen_count == 2:
                    nums.pop(i)
                    continue
                last_seen_count += 1
            else:
                last_seen = num
                last_seen_count = 1
            i += 1
        return len(nums)

s = Solution()
s.removeDuplicates([1,1,1,2,2,3])

File: removeDuplicatesFromSortedArray2.java
// Time Complexity :O(N)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : yes

// Your code here along with comments explaining your approach


public int removeDuplicates(int[] nums) {
    if(nums == null || nums.length==0) return 0;
    
    //using 2 pointers
    int i=1;
    int j=1;
    //for keeping of count of elements
    int count=1;
    
    while(i<nums.length){
        //using i to iterate and check next element
        // if the current element is same as previous then increament the count
        if(nums[i]==nums[i-1]){
            count++;
        }else{ // if the current element is not as previous that means new element and set the count as 1 for new element
            count=1;
        }
        //if count is less than 2, it means we need to swap elements and increase the j counter
        if(count<=2){
            nums[j]=nums[i];
            j++;
        }
        //always icnrease the i pointer
        i++;
    }
    return j;
}